src/Features/Core/Portable/SemanticSearch/Tools/ReferencingSyntaxFinder.cs

```diff
@@ -0,0 +1,124 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#if NET6_0_OR_GREATER
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Threading;
+using System.Threading.Channels;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.FindSymbols;
+using Microsoft.CodeAnalysis.FindUsages;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Microsoft.CodeAnalysis.Shared.Utilities;
+using Microsoft.CodeAnalysis.Threading;
+
+namespace Microsoft.CodeAnalysis.SemanticSearch;
+
+internal sealed class ReferencingSyntaxFinder(Solution solution, CancellationToken cancellationToken)
+{
+    private static readonly FindReferencesSearchOptions s_options = new()
+    {
+        AssociatePropertyReferencesWithSpecificAccessor = false,
+        Cascade = false,
```

Comment:
i think this is fine *for now*.  but we may ahve to reconsider this in the future.  Thsi will prevent finding a call to a virtual that goes through a call to an override (and vice versa).